electrumx/lib/atomicals_blueprint_builder.py

```diff
@@ -400,6 +400,29 @@ def calculate_nft_atomicals_splat(cls, nft_atomicals, tx):
             output_colored_map[expected_output_index]['atomicals'][atomical_id] = atomical_summary_info
             expected_output_index_incrementing += 1
         return AtomicalNftOutputBlueprintAssignmentSummary(output_colored_map)
+    
+    @classmethod
+    def custom_color_nft_atomicals(cls, nft_atomicals, operations_found_at_inputs, tx):
+      # define op `z` to custom-color nft
+      output_colored_map = {}
+      for atomical_id, atomical_info in sorted(nft_atomicals.items()):
+        for out_idx, txout in enumerate(tx.outputs):
+          compact_atomical_id = location_id_bytes_to_compact(atomical_id)
+          compact_atomical_id_data = operations_found_at_inputs["payload"].get(compact_atomical_id, {})
+          # if payload try to color two or more outputs, it will try to color output 0.
+          if len(compact_atomical_id_data.keys()) > 1:
+            expected_output_index = 0
+          else:
+            # if out_idx not in payload keys, skip
+            if str(out_idx) not in compact_atomical_id_data.keys():
```

Comment:
What happens to the nft if it's not specified and skipped?

Reply:
If not specified defined, it will try to color output0. If defined, and the quantity is set to 0 or a negative number, it will burn.